web/packages/teleterm/src/ui/DocumentAccessRequests/NewRequest/NewRequest.tsx

```diff
@@ -17,86 +17,35 @@
  */
 
 import styled from 'styled-components';
+import { Alert, Box, Link } from 'design';
 
-import { Alert, Box, Flex, Link, Indicator } from 'design';
-import { space, SpaceProps, width } from 'design/system';
-
-import {
-  ShowResources,
-  Cluster,
-} from 'gen-proto-ts/teleport/lib/teleterm/v1/cluster_pb';
-
-import { SearchPagination, SearchPanel } from 'shared/components/Search';
+import { ShowResources } from 'gen-proto-ts/teleport/lib/teleterm/v1/cluster_pb';
 import {
   ResourceList,
   ResourceMap,
 } from 'shared/components/AccessRequests/NewRequest';
 
-import {
-  PendingAccessRequest,
-  extractResourceRequestProperties,
-} from 'teleterm/ui/services/workspacesService/accessRequestsService';
+import { PendingAccessRequest } from 'teleterm/ui/services/workspacesService/accessRequestsService';
 
 import { useWorkspaceContext } from 'teleterm/ui/Documents';
 import { useAppContext } from 'teleterm/ui/appContextProvider';
 
-import useNewRequest, { ResourceKind } from './useNewRequest';
-
-const agentOptions: ResourceOption[] = [
-  { value: 'role', label: 'Roles' },
-  {
-    value: 'node',
-    label: 'Servers',
-  },
-  {
-    value: 'app',
-    label: 'Apps',
-  },
-  {
-    value: 'db',
-    label: 'Databases',
-  },
-  {
-    value: 'kube_cluster',
-    label: 'Kubes',
-  },
-];
-
 export function NewRequest() {
   const ctx = useAppContext();
 
-  const { rootClusterUri } = useWorkspaceContext();
-  const rootCluster = ctx.clustersService.findCluster(rootClusterUri);
-  if (rootCluster.showResources === ShowResources.UNSPECIFIED) {
-    return <Indicator />;
-  }
-  return <Inner rootCluster={rootCluster} />;
-}
-
-function Inner(props: { rootCluster: Cluster }) {
   const {
-    attempt,
-    agentFilter,
-    pageCount,
-    updateQuery,
-    updateSearch,
-    selectedResource,
-    customSort,
-    fetch,
-    fetchStatus,
-    onAgentLabelClick,
-    addedResources,
-    addOrRemoveResource,
-    updateResourceKind,
-    prevPage,
-    requestableRoles,
-    nextPage,
-    agents,
-    addedItemsCount,
-  } = useNewRequest(props.rootCluster);
-  const { documentsService, localClusterUri } = useWorkspaceContext();
+    rootClusterUri,
+    localClusterUri,
+    documentsService,
+    accessRequestsService,
+  } = useWorkspaceContext();
+  const rootCluster = ctx.clustersService.findCluster(rootClusterUri);
+  ctx.clustersService.useState();
 
-  const requestStarted = addedItemsCount > 0;
+  const loggedInUser = rootCluster?.loggedInUser;
```

Comment:
Isn't `rootCluster` guaranteed to be not nil here?

Reply:
Hmm I don't think so. I didn't find a parent that wouldn't render its children when there's no root cluster for the given uri.